debezium-server-voyager/debezium-server-voyagerexporter/src/main/java/io/debezium/server/ybexporter/ExportStatus.java

```diff
@@ -284,6 +284,49 @@ public void setSourceType(String sourceType) {
         this.sourceType = sourceType;
     }
 
+    public long getLastQueueSegmentIndex() {
+        Statement selectStmt;
+        int lastSegmentIndex;
+        try {
+            selectStmt = metadataDBConn.createStatement();
+            // Get only 1 result
+            ResultSet rs = selectStmt
+                    .executeQuery(String.format("SELECT segment_no from %s ORDER BY segment_no DESC LIMIT 1",
+                            QUEUE_SEGMENT_META_TABLE_NAME));
+            // if no results are obtained from the query, return -1
+            if (!rs.next()) {
+                return -1;
+            }
+            lastSegmentIndex = rs.getInt("segment_no");
+            selectStmt.close();
+        } catch (SQLException e) {
+            throw new RuntimeException(String.format("Failed to get last queue segment index"), e);
+        }
+        return lastSegmentIndex;
+    }
+
+    public boolean checkIfLastQueueSegmentHasBeenArchivedOrDeleted() {
```

Comment:
nit: you can take segment_no as input just to make the function more flexible and potentially useful in the future.

Reply:
Done